This is an EDA notebook for analyzing the audio data

In [17]:
# Import necessary packages

import os
import tensorflow as tf
import tensorflow_io as tfio
import matplotlib.pyplot as plt
from glob import glob
from tqdm import tqdm

In [18]:
def load_mp3(fpath):
    audio = tfio.audio.AudioIOTensor(fpath)

    return audio

    #print(audio)

    # mp3 = tfio.audio.decode_mp3(byte_contents)

    # return mp3

In [26]:
# Get the sample rates of each mp3
files = glob("./data/fma_medium/**/*.mp3")
rates = []
for file in tqdm(files):
    mp3 = load_mp3(file)
    rates.append(mp3.rate.numpy())

# This will give the unique sample rate values
print(set(rates))

100%|██████████| 25000/25000 [00:11<00:00, 2151.69it/s]

{48000, 32000, 22050, 44100}


In [28]:
for rate in set(rates):
    print(f"Count of sample rate {rate}: {rates.count(rate)}")

Count of sample rate 48000: 1090
Count of sample rate 32000: 17
Count of sample rate 22050: 38
Count of sample rate 44100: 23855


We can see from above that there are 4 different sample rates in our data:
48000, 32000, 22050, and 44100.

Since 32000 and 22050 only have 17 and 38 respectively, I'm comfortable with removing these.
This allows me to avoid downsampling everything else for consistency (although this isn't strictly future-proof)

TODO: Think over the above decision...